### PR TITLE
[FS-507] Support for Remote Welcome Messages

### PR DESCRIPTION
**Pull request by mdimjasevic**
The PR adds support for remote welcome messages. So far only local welcome messages have been supported.

Tracked by https://wearezeta.atlassian.net/browse/FS-507.

## Checklist

 - [x] The **PR Title** explains the impact of the change.
 - [x] The **PR description** provides context as to why the change should occur and what the code contributes to that effect. This could also be a link to a JIRA ticket or a Github issue, if there is one.
 - [x] **changelog.d** contains the following bits of information ([details](https://github.com/wireapp/wire-server/blob/develop/docs/developer/changelog.md)):
   - [x] A file with the changelog entry in one or more suitable sub-sections. The sub-sections are marked by directories inside `changelog.d`.

### CODE REVIEW

File: services/galley/src/Galley/API/MLS/Welcome.hs [hunk 1]
```diff
@@ -102,5 +108,17 @@ sendLocalWelcomes con now rawWelcome lclients = do
...
+  Remote [(UserId, ClientId)] ->
+  Sem r ()
+sendRemoteWelcomes rawWelcome rClients = do
+  let req =
+        MLSWelcomeRequest
+          { mwrRawWelcome = Base64ByteString rawWelcome,
+            mwrRecipients = MLSWelcomeRecipient <$> tUnqualified rClients
+          }
+  let rpc = fedClient @'Galley @"mls-welcome" req
+  -- TODO(md): do error handling
```
Comment (smatting) [inline services/galley/src/Galley/API/MLS/Welcome.hs:123]: @mdimjasevic what errors should we handle here?

Comment (mdimjasevic) [reply services/galley/src/Galley/API/MLS/Welcome.hs:123]: Now that I think about it, I don't see any left. We handle errors in dereferencing a key package reference even before getting into this function, so I think there's no errors left. So feel free to drop that note unless you see any error cases we've forgotten to handle.

File: libs/types-common/src/Data/Json/Util.hs [hunk 2]
```diff
@@ -212,9 +211,6 @@ base64SchemaN = toBase64Text .= parsedText "Base64ByteString" fromBase64Text
 base64Schema :: ValueSchema SwaggerDoc ByteString
 base64Schema = unnamed base64SchemaN
 
-toBase64ByteString :: ByteString -> Base64ByteString
```
Comment (smatting) [inline libs/types-common/src/Data/Json/Util.hs]: So apparently it was being used :)

Comment (mdimjasevic) [reply libs/types-common/src/Data/Json/Util.hs]: Oh, my bad. I used it in a test, but now we know it was a wrong thing to do. I replaced its usage with the `Base64ByteString` data constructor.

**Review by stephen-smith — APPROVED** [review]

**Pull request merged**